lib/plugin.js

```diff
@@ -230,6 +233,21 @@ internals.Plugin.prototype.register = function (plugins /*, [options], callback
             connection._registrations[item.name] = item;
         }
 
+
+        if (item.dependencies.length) {
+
+            // Declare dependencies
+
+            selection.dependency(item.dependencies, function (server, next) {
+
+                // Deferred registration when all dependencies are resolve
```

Comment:
I don't think there's an order to have here, dependencies in registration all just have to be there before the plugin begins his own registration. How is that a change if there wasn't any mechanism for that before ?